File: app/src/main/java/com/ivanovsky/passnotes/presentation/newdb/NewDatabaseContract.java
package com.ivanovsky.passnotes.presentation.newdb;

import com.ivanovsky.passnotes.data.entity.FileDescriptor;
import com.ivanovsky.passnotes.presentation.core.BasePresenter;
import com.ivanovsky.passnotes.presentation.core.BaseView;

class NewDatabaseContract {

	interface View extends BaseView<Presenter> {
		void showError(String message);
		void setStorage(String type, String path);
		void setDoneButtonVisible(boolean visible);
		void showGroupsScreen();
		void hideKeyboard();
		void showStorageScreen();
	}

	interface Presenter extends BasePresenter {
		void createNewDatabaseFile(String filename, String password);
		void selectStorage();
		void onStorageSelected(FileDescriptor file);
	}
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/presentation/newdb/NewDatabaseContract.kt
package com.ivanovsky.passnotes.presentation.newdb

import androidx.lifecycle.LiveData
import com.ivanovsky.passnotes.data.entity.FileDescriptor
import com.ivanovsky.passnotes.presentation.core.BasePresenter
import com.ivanovsky.passnotes.presentation.core.BaseView
import com.ivanovsky.passnotes.presentation.core.livedata.SingleLiveEvent

class NewDatabaseContract {

    interface View : BaseView<Presenter> {
        fun setStorageTypeAndPath(type: String, path: String)
        fun setDoneButtonVisibility(isVisible: Boolean)
        fun showGroupsScreen()
        fun showStorageScreen()
    }

    interface Presenter : BasePresenter {
        val storageTypeAndPath: LiveData<Pair<String, String>>
        val doneButtonVisibility: LiveData<Boolean>
        val showGroupsScreenEvent: SingleLiveEvent<Void>
        val showStorageScreenEvent: SingleLiveEvent<Void>

        fun createNewDatabaseFile(filename: String, password: String)
        fun selectStorage()
        fun onStorageSelected(selectedFile: FileDescriptor)
    }
}